hpx/runtime/applier/applier.hpp

```diff
@@ -159,7 +159,8 @@ namespace hpx { namespace applier
         }
 
         /// Schedule  threads based on the given parcel
-        void schedule_action(parcelset::parcel p, std::size_t num_thread = -1);
+        void schedule_action(parcelset::parcel p, std::size_t num_thread =
+                             (std::numeric_limits<std::size_t>::max)());
```

Comment:
I still think that `std::size_t(-1)` is conveying the intention better than `numeric_limits::max()`. The former is clearly defining a sentinel value, while the latter implies `max()` which is just confusing in this context.